From 51033baa533c248dff7ea37525d1e2455acc96b3 Mon Sep 17 00:00:00 2001
From: Pedro Palazon Candel <pedro@joyent.com>
Date: Fri, 13 Sep 2019 20:21:53 +0200
Subject: [PATCH] TRITON-1894
 svc:/smartdc/agent/firewaller-config-migration:default goes into maint

---
 config-migration.js | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/config-migration.js b/config-migration.js
index c1f5644..7c34093 100644
--- a/config-migration.js
+++ b/config-migration.js
@@ -77,7 +77,17 @@ vasync.pipeline({
                 func: function testIPFFiles(vm, nextVm) {
                     var ipfv4 = util.format(IPF_CONF,
                         ctx.vmsByVmUUID[vm.uuid].zonepath);
-                    var ipfv4Data = fs.readFileSync(ipfv4, 'utf8');
+                    var ipfv4Data;
+                    try {
+                        ipfv4Data = fs.readFileSync(ipfv4, 'utf8');
+                    } catch (err) {
+                        if (err.code !== 'ENOENT') {
+                            nextVm(err);
+                            return;
+                        }
+                        nextVm();
+                        return;
+                    }
                     var ipfv4Res = IPF_VER_RE.exec(ipfv4Data);
                     // If we cannot find a version written on the rules file,
                     // let's assume it's version 1 (pre RFD 163):
-- 
2.21.0

